picard/ui/playertoolbar.py

```diff
@@ -357,38 +346,50 @@ def event(self, event):
         return super().event(event)
 
 
-class PlaybackRatePopover(QtWidgets.QFrame):
-    value_changed = QtCore.pyqtSignal(float)
+class VolumeControlButton(QtWidgets.QToolButton):
+    volume_changed = QtCore.pyqtSignal(int)
 
-    def __init__(self, parent, value):
+    def __init__(self, parent, volume):
         super().__init__(parent)
-        self.setWindowFlags(QtCore.Qt.Popup | QtCore.Qt.FramelessWindowHint)
+        icon = self.style().standardIcon(QtWidgets.QStyle.SP_MediaVolume)
+        self.setIcon(icon)
+        self.set_volume(volume)
+        self.setText('100%')  # Only for initial min width calculation
+        self.clicked.connect(self.show_popover)
 
-        vbox = QtWidgets.QVBoxLayout(self)
-        speed_label = QtWidgets.QLabel(_("Playback speed"), self)
-        speed_label.setAlignment(QtCore.Qt.AlignCenter)
-        vbox.addWidget(speed_label)
-        speed_slider = QtWidgets.QSlider(self)
-        speed_slider.setOrientation(QtCore.Qt.Horizontal)
-        # In 0.1 steps from 0.5 to 1.5
-        multiplier = 10.0
-        speed_slider.setMinimum(5)
-        speed_slider.setMaximum(15)
-        speed_slider.setSingleStep(1)
-        speed_slider.setPageStep(1)
-        speed_slider.setTickInterval(1)
-        speed_slider.setTickPosition(QtWidgets.QSlider.TicksBelow)
-        speed_slider.setValue(value * multiplier)
-        speed_slider.valueChanged.connect(lambda v: self.value_changed.emit(v / multiplier))
-        vbox.addWidget(speed_slider)
+    def show_popover(self):
+        popover = SliderPopover(self, _('Volume'), self.volume)
+        popover.slider.setMinimum(0)
+        popover.slider.setMaximum(100)
+        popover.value_changed.connect(self.on_slider_value_changed)
+        popover.show()
 
-    def show(self):
-        super().show()
-        self._update_position()
+    def on_slider_value_changed(self, value):
+        self.set_volume(value)
+        self.volume_changed.emit(self.volume)
 
-    def _update_position(self):
-        parent = self.parent()
-        x = -(self.width() - parent.width()) / 2
-        y = -self.height()
-        pos = parent.mapToGlobal(QtCore.QPoint(x, y))
-        self.move(pos)
+    def set_volume(self, volume):
+        self.volume = volume
+        label = _('%d%%') % volume
+        self.setText(label)
+
+    def showEvent(self, event):
+        super().showEvent(event)
+        self.setMinimumWidth(self.width())
+        self.set_volume(self.volume)
+
+    def event(self, event):
+        if event.type() == QtCore.QEvent.Wheel:
+            delta = event.angleDelta().y()
+            volume = self.volume
+            if delta > 0:
+                volume += 3
```

Comment:
I'd store 3 in a variable (step?).

Reply:
Fixed as part of https://github.com/metabrainz/picard/pull/1231/commits/164c32ff0131de30f5a97634f1a97e5de157f46c . The 3 was chosen because it is the default `pageStep` for `QSlider`. I now made this explicit by setting the variable and also using this in `slider.setPageStep(...)`